File: JAVA/Pertemuan 02/src/id/ac/uph/pbo/pertemuan02/TugasLatihan.java
package id.ac.uph.pbo.pertemuan02;

import java.util.Scanner;

public class TugasLatihan {

    public static void main(String[] args) {
        System.out.println("=========================================");
        System.out.println("TUGAS LATIHAN MANDIRI - PERTEMUAN 02");
        System.out.println("=========================================\n");

        Scanner scanner = new Scanner(System.in);
        
        System.out.println("Pilih menu latihan:");
        System.out.println("1. Cek Bilangan Ganjil/Genap");
        System.out.println("2. Hitung Deret Angka");
        System.out.println("3. Kalkulator Sederhana");
        System.out.print("Pilihan (1-3): ");
        
        int pilihan = scanner.nextInt();
        
        switch (pilihan) {
            case 1:
                cekGanjilGenap(scanner);
                break;
            case 2:
                hitungDeret(scanner);
                break;
            case 3:
                kalkulatorSederhana(scanner);
                break;
            default:
                System.out.println("Pilihan tidak valid!");
        }
        
        scanner.close();
        System.out.println("\nProgram Selesai.");
    }

    // 1. Method untuk cek ganjil genap
    private static void cekGanjilGenap(Scanner sc) {
        System.out.println("\n--- 1. Cek Ganjil / Genap ---");
        System.out.print("Masukkan sebuah angka integer: ");
        int angka = sc.nextInt();
        
        if (angka % 2 == 0) {
            System.out.println("-> " + angka + " adalah bilangan GENAP.");
        } else {
            System.out.println("-> " + angka + " adalah bilangan GANJIL.");
        }
    }

    // 2. Method untuk hitung deret dengan loop (for)
    private static void hitungDeret(Scanner sc) {
        System.out.println("\n--- 2. Hitung Deret Angka ---");
        System.out.print("Masukkan batas maksimal deret: ");
        int batas = sc.nextInt();
        
        int total = 0;
        System.out.print("Deret: ");
        for (int i = 1; i <= batas; i++) {
            System.out.print(i + (i == batas ? "" : " + "));
            total += i;
        }
        System.out.println("\nTotal Jumlah = " + total);
    }

    // 3. Method kalkulator sederhana dengan while
    private static void kalkulatorSederhana(Scanner sc) {
        System.out.println("\n--- 3. Kalkulator Sederhana ---");
        System.out.print("Masukkan angka pertama: ");
        double num1 = sc.nextDouble();
        System.out.print("Masukkan angka kedua: ");
        double num2 = sc.nextDouble();
        
        System.out.print("Pilih operasi (+, -, *, /): ");
        char op = sc.next().charAt(0);
        
        double hasil = 0;
        boolean valid = true;
        
        switch(op) {
            case '+': hasil = num1 + num2; break;
            case '-': hasil = num1 - num2; break;
            case '*': hasil = num1 * num2; break;
            case '/': 
                if (num2 != 0) hasil = num1 / num2; 
                else {
                    System.out.println("Error: Pembagian dengan nol tidak diizinkan.");
                    valid = false;
                }
                break;
            default:
                System.out.println("Operasi tidak dikenal!");
                valid = false;
        }
        
        if(valid) {
            System.out.printf("Hasil: %.2f %c %.2f = %.2f\n", num1, op, num2, hasil);
        }
    }
}

File: PYTHON/Pertemuan 02/TugasLatihan.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
def cek_ganjil_genap():
    print("\n--- 1. Cek Ganjil / Genap ---")
    try:
        angka = int(input("Masukkan sebuah angka integer: "))
        if angka % 2 == 0:
            print(f"-> {angka} adalah bilangan GENAP.")
        else:
            print(f"-> {angka} adalah bilangan GANJIL.")
    except ValueError:
        print("Input tidak valid.")

def hitung_deret():
    print("\n--- 2. Hitung Deret Angka ---")
    try:
        batas = int(input("Masukkan batas maksimal deret: "))
        total = 0
        print("Deret: ", end="")
        for i in range(1, batas + 1):
            print(f"{i}", end=("" if i == batas else " + "))
            total += i
        print(f"\nTotal Jumlah = {total}")
    except ValueError:
        print("Input tidak valid.")

def kalkulator_sederhana():
    print("\n--- 3. Kalkulator Sederhana ---")
    try:
        num1 = float(input("Masukkan angka pertama: "))
        num2 = float(input("Masukkan angka kedua: "))
        op = input("Pilih operasi (+, -, *, /): ")
        
        hasil = 0.0
        valid = True
        
        if op == '+':
            hasil = num1 + num2
        elif op == '-':
            hasil = num1 - num2
        elif op == '*':
            hasil = num1 * num2
        elif op == '/':
            if num2 != 0:
                hasil = num1 / num2
            else:
                print("Error: Pembagian dengan nol tidak diizinkan.")
                valid = False
        else:
            print("Operasi tidak dikenal!")
            valid = False
        
        if valid:
            print(f"Hasil: {num1:.2f} {op} {num2:.2f} = {hasil:.2f}")
    except ValueError:
        print("Input tidak valid.")

def main():
    print("=========================================")
    print("TUGAS LATIHAN MANDIRI - PERTEMUAN 02")
    print("=========================================\n")

    print("Pilih menu latihan:")
    print("1. Cek Bilangan Ganjil/Genap")
    print("2. Hitung Deret Angka")
    print("3. Kalkulator Sederhana")
    
    try:
        pilihan = int(input("Pilihan (1-3): "))
        if pilihan == 1:
            cek_ganjil_genap()
        elif pilihan == 2:
            hitung_deret()
        elif pilihan == 3:
            kalkulator_sederhana()
        else:
            print("Pilihan tidak valid!")
    except ValueError:
        print("Pilihan tidak valid!")
        
    print("\nProgram Selesai.")

if __name__ == "__main__":
    main()

